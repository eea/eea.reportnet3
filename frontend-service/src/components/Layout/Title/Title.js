import React from 'react';
import styles from './Title.module.css';
import dsIcon from '../../../assets/images/dataset_icon.png';

const Title = (props) => {
    return (      
      <div className={styles.Title}>
        <h2><img  src={dsIcon} alt="Dataset"/>{props.title}</h2>
      </div>  
    );
}

export default React.memo(Title);