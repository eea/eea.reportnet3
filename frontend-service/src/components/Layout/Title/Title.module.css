.Title {
  text-align: center;
  color: #004494;
}