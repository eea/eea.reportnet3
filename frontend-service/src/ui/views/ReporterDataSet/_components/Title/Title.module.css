.Title {
  color: #404040;
  background-color: #ffffff;
  display: table-cell;
  width: 100%;
}

.Title h2 {
  font-weight: 50;
  vertical-align: middle;
  text-align: center;
}
.Title img {
  height: 35px;
  vertical-align: middle;
  text-align: center;
  margin-right: 10px;
}
