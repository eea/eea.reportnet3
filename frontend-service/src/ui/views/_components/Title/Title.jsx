import React from 'react';

import styles from './Title.module.css';

import { AwesomeIcons } from 'conf/AwesomeIcons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Title = React.memo(({ title, icon }) => {
  return (
    <div className={styles.Title}>
      <h2>
        <FontAwesomeIcon icon={AwesomeIcons(icon)} style={{ marginRight: '0.3rem' }} />
        {title}
      </h2>
    </div>
  );
});

export { Title };
