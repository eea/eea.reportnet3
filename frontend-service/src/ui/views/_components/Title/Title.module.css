.titleWrap {
  align-content: center;
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  margin: 1rem 1rem 1rem 0rem;
}

.title {
  display: table-cell;
  font-weight: 100;
}

.subtitle {
  color: var(--c-corporate-blue);
  font-weight: 100;
  margin: 0rem !important;
}

.icon {
  color: var(--gray-50);
  margin-right: 0.5rem !important;
}
