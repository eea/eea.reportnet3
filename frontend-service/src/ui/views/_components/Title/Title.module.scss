.titleWrap {
  align-content: center;
  align-items: center;
  display: flex;
  margin: 2rem 0rem 2rem 0rem;
}

.iconWrap {
  align-content: center;
}

.title {
  display: table-cell;
  font-size: 2.25rem;
  color: var(--title-color);
  margin: 0px;
  font-weight: bold;

  span {
    font-size: 1.3rem;
  }
}

.subtitle {
  font-size: 1.3rem;
  color: var(--subtitle-color);
  margin: 0px;
  font-weight: 400;
}

.icon {
  color: var(--title-icon-color);
  opacity: 0.6;
  margin-right: 0.5rem !important;
}
