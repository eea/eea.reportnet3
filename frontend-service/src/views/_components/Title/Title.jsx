import { memo } from 'react';

import { AwesomeIcons } from 'conf/AwesomeIcons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import styles from './Title.module.scss';

export const Title = memo(({ icon, iconSize, insideTitle, subtitle, title }) => (
  <div className={styles.titleWrap}>
    <div className={styles.iconWrap}>
      <FontAwesomeIcon
        className={styles.icon}
        icon={AwesomeIcons(icon)}
        role="presentation"
        style={{ fontSize: iconSize }}
      />
    </div>
    <div className={styles.textWrap}>
      <h1 className={styles.title}>
        {title}
        <span>{insideTitle}</span>
      </h1>
      <h3 className={styles.subtitle}>{subtitle}</h3>
    </div>
  </div>
));
