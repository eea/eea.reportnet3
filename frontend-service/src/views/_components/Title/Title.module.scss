.titleWrap {
  align-content: center;
  align-items: center;
  display: flex;
  margin: 2rem 0rem 2rem 0rem;
}

.iconWrap {
  align-content: center;
}

.title {
  display: table-cell;
  font-size: 2.25rem;
  color: var(--title-color);
  margin: 0px;
  font-weight: bold;

  span {
    font-size: 1.3rem;
    font-style: italic;
    margin-left: 1rem;
  }
}

.subtitle {
  font-size: 1.3rem;
  color: var(--subtitle-color);
  margin: 0px;
  font-weight: 400;
}

.icon {
  color: var(--title-icon-color);
  opacity: var(--title-icon-opacity);
  margin-right: 0.5rem !important;
}
