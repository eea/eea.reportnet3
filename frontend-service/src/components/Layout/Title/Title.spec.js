import React from 'react';
import { shallow } from 'enzyme';
import Title from './Title';

describe('Title', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<Title />));
    
    //Snapshot
    it('should render correctly', () => expect(wrapper).toMatchSnapshot());
  
    it('should render a <div />', () => {
      expect(wrapper.find('div').length).toEqual(1);
    });

  });