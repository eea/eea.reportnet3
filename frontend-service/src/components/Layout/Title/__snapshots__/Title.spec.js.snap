// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title should render correctly 1`] = `ShallowWrapper {}`;
